miniapps/toys/life.cpp

```diff
@@ -0,0 +1,348 @@
+// Copyright (c) 2010, Lawrence Livermore National Security, LLC. Produced at
+// the Lawrence Livermore National Laboratory. LLNL-CODE-443211. All Rights
+// reserved. See file COPYRIGHT for details.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability see http://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the GNU Lesser General Public License (as published by the Free
+// Software Foundation) version 2.1 dated February 1999.
+//
+//               ----------------------------------------
+//               Life Miniapp:  Model of the Game of Life
+//               ----------------------------------------
+//
+// This miniapp implements Conway's Game of Life.  A few simple starting
+// positions are available as well as a default random initial state.  The
+// game will terminate only if two successive iterations are identical.
+//
+// See the output of 'life -h' for more options.
+//
+// Compile with: make life
+//
+// Sample runs: life
+//              life -nx 100 -ny 100 -r 0.3
+
```

Comment:
Done.  I added a default initialization using either 'M' or, space permitting, the characters "MFEM".  It's a bit hard to see because they fly by so quickly but we can consider it an easter egg for our devoted fans.  I also added a couple examples using the "sketchpad", "glider", and "blinker" options.

Reply:
Very nice, thanks Mark!